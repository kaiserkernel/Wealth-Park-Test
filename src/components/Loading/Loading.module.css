.loading {
  @apply border-t-transparent w-10 h-10 border-4 border-blue-400 border-solid rounded-full animate-spin;
}
