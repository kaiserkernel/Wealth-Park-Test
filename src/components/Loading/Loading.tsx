/* eslint-disable react/require-default-props */
import { FC } from 'react'
import classNames from 'classnames'

import styles from './Loading.module.css'

interface Props {
  loadingText?: string
  className?: string
}

const Loading: FC<Props> = ({ loadingText, className }) => {
  return (
    <div className={classNames('flex-center space-x-4', className)} data-testid="loading">
      <div className={styles.loading} />
      {loadingText && <span> {loadingText} </span>}
    </div>
  )
}

export default Loading
