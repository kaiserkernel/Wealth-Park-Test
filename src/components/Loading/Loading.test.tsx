import { render, screen } from '@testing-library/react'

import Loading from './Loading'

describe('Post List Card', () => {
  const LOADING = {
    loadingText: 'Loading',
    className: '',
  }

  it('renders correctly', () => {
    const { container } = render(<Loading {...LOADING} />)
    expect(container).toMatchSnapshot()
  })

  it('renders Loading', () => {
    render(<Loading {...LOADING} />)

    const loading = screen.getByTestId('loading')

    expect(loading).toBeInTheDocument()
  })
})
