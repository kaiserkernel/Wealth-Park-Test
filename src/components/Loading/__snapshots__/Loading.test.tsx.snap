// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post List Card renders correctly 1`] = `
<div>
  <div
    class="flex-center space-x-4"
    data-testid="loading"
  >
    <div
      class="loading"
    />
    <span>
       
      Loading
       
    </span>
  </div>
</div>
`;
